yt_dlp/extractor/twitter.py

```diff
@@ -552,10 +701,84 @@ class TwitterIE(TwitterBaseIE):
+            raise ExtractorError(f'Twitter API says: {cause}', expected=True)
+
+        status = result.get('legacy', {})
+        status.update(traverse_obj(result, {
+            'user': ('core', 'user_results', 'result', 'legacy'),
+            'card': ('card', 'legacy'),
+            'quoted_status': ('quoted_status_result', 'result', 'legacy'),
+        }, expected_type=dict, default={}))
+
+        # extra transformations are needed bc LeGaCy cOmPaTiBiLiTy
+        binding_values = {
+            binding_value.get('key'): binding_value.get('value')
+            for binding_value in traverse_obj(status, ('card', 'binding_values', ...), expected_type=dict)
+        }
+        if binding_values:
+            status['card']['binding_values'] = binding_values
+
+        return status
+
+    def _build_graphql_query(self, media_id):
+        return {
+            'variables': {
+                'focalTweetId': media_id,
+                'includePromotedContent': True,
+                'with_rux_injections': False,
+                'withBirdwatchNotes': True,
+                'withCommunity': True,
+                'withDownvotePerspective': False,
+                'withQuickPromoteEligibilityTweetFields': True,
+                'withReactionsMetadata': False,
+                'withReactionsPerspective': False,
+                'withSuperFollowsTweetFields': True,
+                'withSuperFollowsUserFields': True,
+                'withV2Timeline': True,
+                'withVoice': True,
+            },
+            'features': {
+                'graphql_is_translatable_rweb_tweet_is_translatable_enabled': False,
+                'interactive_text_enabled': True,
+                'responsive_web_edit_tweet_api_enabled': True,
+                'responsive_web_enhance_cards_enabled': True,
+                'responsive_web_graphql_timeline_navigation_enabled': False,
+                'responsive_web_text_conversations_enabled': False,
+                'responsive_web_uc_gql_enabled': True,
+                'standardized_nudges_misinfo': True,
+                'tweet_with_visibility_results_prefer_gql_limited_actions_policy_enabled': False,
+                'tweetypie_unmention_optimization_enabled': True,
+                'unified_cards_ad_metadata_container_dynamic_card_content_query_enabled': True,
+                'verified_phone_label_enabled': False,
+                'vibe_api_enabled': True,
+            },
+        }
+
     def _real_extract(self, url):
         twid = self._match_id(url)
-        status = self._call_api(
-            'statuses/show/%s.json' % twid, twid, {
+        use_graphql = self.is_logged_in()
+
+        force_graphql = bool(self._configuration_arg('force_graphql', ie_key=self.ie_key()))
+        if force_graphql and not use_graphql:
+            self.report_warning('Forced usage of GraphQL API', only_once=True)
+            use_graphql = True
```

Comment:
warning is unnecessary. If you need the message, put it in debug/stdout
```suggestion
        if not use_graphql and self._configuration_arg('force_graphql'):
            self.write_debug('Forced usage of GraphQL API')
            use_graphql = True
```